tests/components/wled/test_light.py

```diff
@@ -137,6 +140,40 @@ async def test_switch_change_state(
         )
 
 
+async def test_dynamically_handle_segments(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test if a new/deleted segment is dynamically added/removed."""
+    await init_integration(hass, aioclient_mock)
+
+    assert hass.states.get("light.wled_rgb_light")
+    assert hass.states.get("light.wled_rgb_light_1")
+
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+    entry = entity_registry.async_get("light.wled_rgb_light")
+    coordinator = hass.data[DOMAIN][entry.config_entry_id]
+
+    data = json.loads(load_fixture("wled/rgb_single_segment.json"))
+    device = WLEDDevice(data)
+
+    # Test removal if segment went missing
+    with patch(
+        "homeassistant.components.wled.WLEDDataUpdateCoordinator._async_update_data",
+        return_value=device,
+    ):
+        await coordinator.async_refresh()
```

Comment:
Can we move time forward instead, by patching it and firing a time changed event?

Reply:
Here's a test that does this:
https://github.com/home-assistant/core/blob/465b98513b20a8e8f030cf795dd72f559f984a66/tests/components/gdacs/test_geo_location.py#L43